packages/eslint-plugin/src/rules/dot-notation.ts

```diff
@@ -0,0 +1,80 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import * as ts from 'typescript';
+import baseRule from 'eslint/lib/rules/dot-notation';
+import {
+  InferOptionsTypeFromRule,
+  InferMessageIdsTypeFromRule,
+  createRule,
+  getParserServices,
+} from '../util';
+
+export type Options = InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default createRule<Options, MessageIds>({
+  name: 'dot-notation',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce dot notation whenever possible',
+      category: 'Best Practices',
+      recommended: false,
+      extendsBaseRule: true,
+      requiresTypeChecking: true,
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowKeywords: {
+            type: 'boolean',
+            default: true,
+          },
+          allowPattern: {
+            type: 'string',
+            default: '',
+          },
+          allowPrivateClassPropertyAccess: {
+            tyoe: 'boolean',
+            default: true,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    fixable: baseRule.meta.fixable,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: [
+    {
+      allowPrivateClassPropertyAccess: true,
```

Comment:
please set it to default false - not everyone will want to allow this out of the box.